File: Bypyhton/盛最多水的容器.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        left, right = 0, len(height) - 1
        maxV = 0
        while left < right:
            h = height[left] if height[left] < height[right] else height[right]
            w = right - left
            v = h * w
            if v > maxV:
                maxV = v
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return maxV

File: ByCpp/盛最多水的容器.cpp
#include <vector>
#include <algorithm>

using namespace std;

int maxArea(vector<int>& height) {
    int left = 0, right = height.size() - 1;
    int maxV = 0;
    while (left < right) {
        int h = min(height[left], height[right]);
        int w = right - left;
        maxV = max(maxV, h * w);
        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }
    return maxV;
}
